File: framework/ts/src/bootstrap.ts
import {
    AvlTree,
    BinarySearchTree
} from '@datastructures-js/binary-search-tree';
import {
    PriorityQueue,
    MinPriorityQueue,
    MaxPriorityQueue
} from '@datastructures-js/priority-queue';
import { ListNode, NestedInteger } from './soda/leetcode';
import { TestWork, Utils, Validators } from './soda/unittest';

// step [1]: implement solution function
/**
 * @param {number} a
 * @param {number} b
 * @return {number}
 */
function add(a: number, b: number): number {
    return a + b;
}

// step [2]: setup function/return/arguments
let taskFunc = add;
const work = TestWork.create<ReturnType<typeof taskFunc>>(taskFunc);
// let work = TestWork.forStruct(STRUCT);
// work.validator = (x, y) => { ... };
work.compareSerial = true;
console.log(work.run(Utils.fromStdin()));

File: framework/kotlin/src/main/kotlin/bootstrap.kt
import soda.kotlin.unittest.*

class Solution {
    fun add(a: Int, b: Int): Int {
        return a + b
    }
}

class __Bootstrap__ : (String) -> String {
    override fun invoke(text: String): String {
        val work = GenericTestWork.create(Solution()::add)

        // * by method that has not return value
        // val work = GenericTestWork.createVoid(Solution.METHOD_WITHOUT_RETURN)
        // * by class of data struct
        // val work = GenericTestWork.forStruct(STRUCT::class)

        // * setup validator
        // work.validator = (T, T) -> Boolean
        work.compareSerial = true
        return work.run(text)
    }

}

fun main() {
    print(__Bootstrap__()(Utils.fromStdin()))
}
